tests/quality/test_quality.py

```diff
@@ -0,0 +1,123 @@
+import os
+
+import numpy as np
+import pandas as pd
+import pytest
+from sklearn.linear_model import LinearRegression
+from sklearn.model_selection import cross_val_score
+
+from rdt import HyperTransformer
+from rdt.transformers import NumericalTransformer, get_transformers_by_type
+from tests.quality.utils import download_single_table_dataset
+
+THRESHOLD = 0.4
+
+TYPE_TO_DTYPE = {
+    'numerical': ['number'],
+    'float': ['float'],
+    'int': ['int'],
+    'categorical': ['object', 'category'],
+    'datetime': ['datetime'],
+    'boolean': ['bool']
+}
+
+
+def format_array(array):
+    if not isinstance(array, np.ndarray):
+        array = array.to_numpy()
+
+    if len(array.shape) == 1:
+        array = array.reshape(-1, 1)
+    return array
+
+
+def get_regression_score(features, target):
+    model = LinearRegression()
+    scores = cross_val_score(model, features, target)
+    return np.mean(scores)
+
+
+def find_columns(data, data_type):
+    dtypes = TYPE_TO_DTYPE.get(data_type, data_type)
+    columns = set()
+    for dtype in dtypes:
+        selected = data.select_dtypes(dtype)
+        columns.update(set(selected.columns))
+
+    return columns
+
+
+def get_transformer_scores(data, data_type, transformers):
+    columns_to_predict = find_columns(data, 'numerical')
+    columns_to_transform = find_columns(data, data_type)
+    all_scores = pd.DataFrame()
+    index = [transformer.__name__ for transformer in transformers]
+    features = data[columns_to_transform]
+
+    for column in columns_to_predict:
+        target = data[column].to_frame()
+        numerical_transformer = NumericalTransformer(null_column=False)
+        target = numerical_transformer.fit_transform(target, list(target.columns))
+        target = format_array(target)
+        scores = []
+        for transformer in transformers:
+            ht = HyperTransformer(data_type_transformers={data_type: transformer})
+            ht.fit(features)
+            transformed_features = ht.transform(features).to_numpy()
+            scores.append(get_regression_score(transformed_features, target))
+        all_scores[column] = pd.Series(scores, index=index)
+    return all_scores
+
+
+def validate_relative_score(scores, transformer):
+    scores_without_transformer = scores.drop(transformer)
+    means = scores_without_transformer.mean()
+    means_above_threshold = means > THRESHOLD
+    standard_deviations = scores_without_transformer.std()
+    minimum_scores = means[means_above_threshold] - standard_deviations[means_above_threshold]
+
+    assert all(scores.loc[transformer, means_above_threshold] > minimum_scores)
+
+
+def get_test_cases():
+    max_size = 5000000
```

Comment:
nit: can make this (and types_to_skip) a constant at the top of the file if we're going to keep using it